Automattic/wp-calypso
# Scheduled Updats: Add gating component for scheduled updates feature

Related to Automattic/dotcom-forge#5894

## Proposed Changes

* Add a new WPCOM_Features in Calypso feature list.
* Add a new gating component for the scheduled update feature. This will block the UI if a user's site is not an Atomic site or doesn't have the scheduled update feature.

## Testing Instructions



* Use a simple site, navigate to `http://calypso.localhost:3000/plugins/scheduled-updates/<your-site-slug>`
* You should see the page is gated and showing the upsell banner like below:
![Screen Shot 2024-03-07 at 12 15 06 AM](https://github.com/Automattic/wp-calypso/assets/4074459/a0c11322-a034-425f-ab3c-12e8ff6ba5fc)

* Try to navigate to `http://calypso.localhost:3000/plugins/scheduled-updates/create/<your-site-slug>` and you'll see the same.
* Click the `Upgarde` button from the banner and make sure it brings you to the checkout page, it should redirect you back to the scheduled update page after you upgrade the plan.
* Once you're back, you'll see a banner asking you to activate the hosting access.

![Screen Shot 2024-03-07 at 12 16 39 AM](https://github.com/Automattic/wp-calypso/assets/4074459/d3c91786-87b6-4c00-9f0d-5004c59073d6)


* Click `Activate` and you'll see it change to a loading state while waiting for the transfer to finish.
![Screen Shot 2024-03-06 at 1 32 14 AM](https://github.com/Automattic/wp-calypso/assets/4074459/449aa95d-2c2a-4bd5-a1e3-ca667f0b6246)

* Once the transfer is finished the UI should be unblocked.
* Try to navigate to `http://calypso.localhost:3000/plugins/scheduled-updates/<your-site-slug>` directly, and you shouldn't get blocked now.


## Pre-merge Checklist
